spec/integration/extensions/active_record_spec.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+RSpec.describe Dry::Operation::Extensions::ActiveRecord do
+  include Dry::Monads[:result]
+
+  let!(:model) do
+    Class.new(ActiveRecord::Base) do
+      self.table_name = :foo
+    end
+  end
+
+  before :all do
+    ActiveRecord::Base.establish_connection(adapter: "sqlite3", database: ":memory:")
+    ActiveRecord::Schema.define do
+      create_table :foo do |t|
+        t.string :bar
+      end
+    end
+  end
+
+  after :each do
+    model.delete_all
+  end
+
+  let(:base) do
+    Class.new(Dry::Operation) do
+      include Dry::Operation::Extensions::ActiveRecord
+    end
+  end
+
+  it "rolls transaction back on failure" do
+    instance = Class.new(base) do
+      def call
+        transaction do
+          step create_record
+          step failure
+        end
+      end
+
+      def create_record
+        Success(ActiveRecord::Base.descendants.first.create(bar: "bar"))
+      end
+
+      def failure
+        Failure(:failure)
+      end
+    end.new
```

Comment:
```suggestion
    instance = Class.new(base) do
     attr_reader :model
     
      def initialize(model)
        @model = model
      end
      
      def call
        transaction do
          step create_record
          step failure
        end
      end

      def create_record
        Success(model.create(bar: "bar"))
      end

      def failure
        Failure(:failure)
      end
    end.new(model)
```

In this way we can get rid of the `descendants` magic, which from experience it may be brittle (maybe that was the need for `let!` with bang above?)